testcases/OpTestKexec.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import time
+import tempfile
+
+import OpTestConfiguration
+from common import OpTestInstallUtil
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+
+
+class OpTestKexec(unittest.TestCase):
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.cv_BMC = conf.bmc()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.pdbg = conf.args.pdbg
+        self.basedir = conf.basedir
+        self.c = self.cv_SYSTEM.console
+        self.kernel_image = conf.args.kernel_image
+        self.initrd_image = conf.args.initrd_image
+        if self.bmc_type == "FSP_PHYP":
+            self.is_lpar = True
+            self.hmc_user = conf.args.hmc_username
+            self.hmc_password = conf.args.hmc_password
+            self.hmc_ip = conf.args.hmc_ip
+            self.lpar_name = conf.args.lpar_name
+            self.system_name = conf.args.system_name
+            self.cv_HMC = self.cv_SYSTEM.hmc
+            try: self.cpu_resource = conf.args.cpu_resource
+            except AttributeError:
+                self.cpu_resource = 1
+            try: self.mem_resource = conf.args.mem_resource
+            except AttributeError:
+                self.mem_resource = 2048
+        try: self.url = conf.args.url
+        except AttributeError: 
+            self.url = "http://liquidtelecom.dl.sourceforge.net/project/ebizzy/ebizzy/0.3/ebizzy-0.3.tar.gz"
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        res = self.c.run_command("cat /etc/os-release")
+        if "Ubuntu" in res[0] or "Ubuntu" in res[1]:
+            self.distro = "ubuntu"
+        elif 'Red Hat' in res[0] or 'Red Hat' in res[1]:
+            self.distro = 'rhel'
+        elif 'SLES' in res[0] or 'SLES' in res[1]:
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")
+
+    def setup_test(self, dump_place="local"):
+        if dump_place == "local":
+            self.crash_content = self.c.run_command(
+                "ls -l /var/crash | grep '^d'| awk '{print $9}'")
+
+    ##
+    # @brief This function will test the kexec version
+    #
+    def kexec_version(self):
+        self.c.run_command("uname -a")
+        self.c.run_command("cat /etc/os-release")
+        my_cmd = "kexec --version"
+        self.c.run_command(my_cmd)
+        
+    ##
+    # @brief This function will test the kexec load
+    #
+    def kexec_load(self):
+        self.c.run_command("uname -a")
+        self.c.run_command("cat /etc/os-release")
+        my_cmd = "kexec -l /boot/{} --initrd /boot/{} --append=\"`cat /proc/cmdline`\"".format(self.kernel_image,self.initrd_image)
+        self.c.run_command(my_cmd)
```

Comment:
First 2 statements aren't required. Just running kexec -l command will not help. A check should be added to verify if the kernel was loaded successfully. You can use /sys/kernel/kexec_loaded for verification.  We also need code to actually perform kexec operation and verify if kexec boot is successful. 

Can you rework this test and post if for review ?

Reply:
@sacsant as it is tool verification around kexec, so added to verify  all the kexec tool subcommand, yes I do agree this test case need to be rework having added all the option included and kexec -l and verify same, @SACHIN-BAPPALIGE hoping he  is taking care as have offline discussion